### DISCUSSION TITLE
Server Error  ReferenceError: localStorage is not defined

### DISCUSSION THREAD

**Discussion by Efendi5** [neuefische/questions#567] [Web] [answered]
Hallo,
gerade arbeite ich an meinem Branch ResultList, bei refreshen der Seite tritt folgender Fehler auf.
Durch diesen Fehler kann ich das Projekt auch nicht bei Vercel deployn.

Ich habe meinen aktuellen Stand committed und gepusht. [Dieser ist unter diesem Link erreichbar.](https://github.com/Efendi5/capstone-project/tree/ResultList) Der Branch heißt ResultList. 
Der Fehler tritt auf in der Komponente Result: [index.js](https://github.com/Efendi5/capstone-project/blob/ResultList/components/ResultListPage/Result/index.js)

Beschreibung des Fehlers:
Den Localstorage benutze ich um die Punkte beider User in jeder Runde zu speichern, damit ich am Ende TotalPoints ausgeben kann. 
Ein kurzes Code snippet, aber der ganze Code ist dennoch verlinkt:

  const round1Points1 = localStorage.getItem("User1-round1") || 0;
  const round1Points2 = localStorage.getItem("User2-round1") || 0;
  const round2Points1 = localStorage.getItem("User1-round2") || 0;
  const round2Points2 = localStorage.getItem("User2-round2") || 0;
  const round3Points1 = localStorage.getItem("User1-round3") || 0;
  const round3Points2 = localStorage.getItem("User2-round3") || 0;

  useEffect(() => {
    if (currentRound === 1) {
      localStorage.setItem("User1-round1", points1);
      localStorage.setItem("User2-round1", points2);
    }
    if (currentRound === 2) {
      localStorage.setItem("User1-round2", points1);
      localStorage.setItem("User2-round2", points2);
    }
    if (currentRound === 3) {
      localStorage.setItem("User1-round3", points1);
      localStorage.setItem("User2-round3", points2);
    }

Folgende ERROR Message

Server Error

ReferenceError: localStorage is not defined
This error happened while generating the page. Any console logs will be displayed in the terminal window.

Search for the keywords to learn more about each error.

Vielen Dank für eure Hilfe!

**Comment by djfarly** [comment, marked as answer]
Hej @Efendi5,

localStorage is only available on the client. Next.js renders your components on the server and on the client. Thats why you're getting this error message.

I'd suggest you use the `use-local-storage-state` hook, that abstracts localStorage in a way that works with Next.js.

Read more here: https://github.com/neuefische/web-ffm-23-1/blob/main/sessions/react-with-local-storage/react-with-local-storage.md#use-local-storage-state

**Reply by djfarly** [reply]
For later reference here is a direct link to the library: https://github.com/astoilkov/use-local-storage-state

**Reply by Efendi5** [reply]
Danke! Das hat geholfen.
